worlds/cv64/__init__.py

```diff
@@ -0,0 +1,313 @@
+            raise FileNotFoundError(rom_file)
+
+    def generate_early(self) -> None:
+        # Generate the player's unique authentication
+        self.auth = bytearray(self.multiworld.random.getrandbits(8) for _ in range(16))
+
+        self.total_s1s = self.options.total_special1s.value
+        self.s1s_per_warp = self.options.special1s_per_warp.value
+        self.drac_condition = self.options.draculas_condition.value
+
+        # If there are more S1s needed to unlock the whole warp menu than there are S1s in total, drop S1s per warp to
+        # something manageable.
+        while self.s1s_per_warp * 7 > self.total_s1s:
+            self.s1s_per_warp -= 1
+
+        # Set the total and required Special2s to 1 if the drac condition is the Crystal, to the specified YAML numbers
+        # if it's Specials, or to 0 if it's None or Bosses. The boss totals will be figured out later.
+        if self.drac_condition == DraculasCondition.option_crystal:
+            self.total_s2s = 1
+            self.required_s2s = 1
+        elif self.drac_condition == DraculasCondition.option_specials:
+            self.total_s2s = self.options.total_special2s.value
+            self.required_s2s = int(self.options.percent_special2s_required.value / 100 * self.total_s2s)
+        else:
+            self.total_s2s = 0
+            self.required_s2s = 0
+
+        # Enable/disable character stages and branching paths accordingly
+        self.reinhardt_stages = True
+        self.carrie_stages = True
+        self.branching_stages = False
+
+        if self.options.character_stages == CharacterStages.option_reinhardt_only:
+            self.carrie_stages = False
+        elif self.options.character_stages == CharacterStages.option_carrie_only:
+            self.reinhardt_stages = False
+        elif self.options.character_stages == CharacterStages.option_both:
+            self.branching_stages = True
+
+        self.active_stage_exits = get_normal_stage_exits(self)
+
+        stage_1_blacklist = []
+
+        # Prevent Clock Tower from being Stage 1 if more than 4 S1s are needed to warp out of it.
+        if self.s1s_per_warp > 4 and not self.options.multi_hit_breakables:
+            stage_1_blacklist.append(rname.clock_tower)
+
+        # Shuffle the stages if the option is on.
+        if self.options.stage_shuffle:
+            self.active_stage_exits, self.starting_stage, self.active_stage_list = \
+                shuffle_stages(self, stage_1_blacklist)
+        else:
+            self.starting_stage = rname.forest_of_silence
+            self.active_stage_list = [stage for stage in vanilla_stage_order if stage in self.active_stage_exits]
+
+        # Create a list of warps from the active stage list. They are in a random order by default and will never
+        # include the starting stage.
+        self.active_warp_list = generate_warps(self)
+
+    def create_regions(self) -> None:
+        # Add the Menu Region.
+        reg_names = ["Menu"]
```

Comment:
ironically i think you actually got rid of the bit of code where i said that get_region was better to use XD so with how you're doing it now, it's better to just make this a list of regions and then iterate over that to do all your other stuff instead of the list of names, since you won't need to do any lookup at all that way.